src/uint/bits.rs

```diff
@@ -1,6 +1,13 @@
 use crate::{Limb, UInt, Word};
 
 impl<const LIMBS: usize> UInt<LIMBS> {
+    /// Get the value of the bit at position `index`, as a 0- or 1-valued Word.
+    #[inline]
+    pub const fn bit(self, index: usize) -> Word {
+        assert!(index < LIMBS * Limb::BIT_SIZE);
```

Comment:
I think there are a few different ways to avoid panicking in this case:

- Return `0`. This is the [behavior of `num-bigint`](https://docs.rs/num-bigint/0.4.3/src/num_bigint/biguint.rs.html#922-931).
- Return `Option` or `Result`, which would still enable detecting the index is out of range in `const fn` contexts

Reply:
I prefer the first option if panicking is not desired. I think that generally the index will be known to be in a valid range (for example when used in `for idx in 0..UInt::BIT_SIZE`) and so the assert just gets optimized out.

Reply:
First option seems good to me too

Reply:
Yep agreed